process/block/baseProcess_test.go

```diff
@@ -808,6 +810,22 @@ func TestCheckProcessorNilParameters(t *testing.T) {
 			},
 			expectedErr: state.ErrNilAccountFactory,
 		},
+		{
+			args: func() blproc.ArgBaseProcessor {
+				args := createArgBaseProcessor(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+				args.RunTypeComponents = &mock.RunTypeComponentsStub{AccountCreator: &stateMock.AccountsFactoryStub{}, DataCodec: nil}
+				return args
+			},
+			expectedErr: errorsMx.ErrNilDataCodec,
+		},
+		{
+			args: func() blproc.ArgBaseProcessor {
+				args := createArgBaseProcessor(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+				args.RunTypeComponents = &mock.RunTypeComponentsStub{AccountCreator: &stateMock.AccountsFactoryStub{}, DataCodec: &sovereign.DataCodecMock{}, TopicsChecker: nil}
```

Comment:
I think here and above we could refactor to use constructor + replace only needed field to be nil like:

```
runTypeComps :=  NewRunTypeComponentsStub()
runTypeComps.TopicsChecker = nil
```